### PR TITLE
Mothur: added macros to all wrappers

### PR DESCRIPTION
**Pull request by oxyko**
In this change I've refactored all mother wrappers to use macros for <requirements>, token text in help ("Mother Overview" section), and added citation.
 `planemo lint` now passes, except for tests.

Part of the effort towards #419

### CODE REVIEW

**Comment by blankenberg** [comment]
@oxyko Some merge conflicts exist. Can you resolve them and update the pull request? Thanks!

**Comment by bgruening** [comment]
This is wonderful! Please let us know if you need help with the rebase?

**Comment by oxyko** [comment]
Merge conflicts are resolved.

File: tools/mothur/tools/mothur/align.check.xml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <tool id="mothur_align_check" name="Align.check" version="1.20.0">
  <description>Calculate the number of potentially misaligned bases</description>
+<macros>
+	<import>macros.xml</import>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/align.check.xml]: Can we use spaces for indentation here and elsewhere?

File: tools/mothur/tools/mothur/consensus.seqs.xml [hunk 2]
```diff
@@ -59,25 +64,20 @@
    <filter>otu['use'] == 'no'</filter>
   </data>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/consensus.seqs.xml]: This line should be removed as well.

**Comment by bgruening** [comment]
@blankenberg ready for merge? Indentation can be a separate PR, right?

File: tools/mothur/tools/mothur/corr.axes.xml [hunk 3]
```diff
@@ -64,19 +69,13 @@
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
   <data format="axes" name="corr_axes" label="${tool.name} on ${on_string}: corr.axes"/>
  </outputs>
- <requirements>
   <requirement type="package" version="1.33" >mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/corr.axes.xml]: Remove this line as well.

File: tools/mothur/tools/mothur/get.otulist.xml [hunk 4]
```diff
@@ -37,19 +42,14 @@
  <outputs>
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/get.otulist.xml]: remove?

File: tools/mothur/tools/mothur/get.oturep.xml [hunk 5]
```diff
@@ -100,19 +105,14 @@
  <outputs>
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/get.oturep.xml]: remove?

File: tools/mothur/tools/mothur/get.rabund.xml [hunk 6]
```diff
@@ -33,19 +38,13 @@
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
   <data format="rabund" name="rabund" label="${tool.name} on ${on_string}: rabund (Rank Abundance)"/>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/get.rabund.xml]: remove

File: tools/mothur/tools/mothur/get.seqs.xml [hunk 7]
```diff
@@ -84,19 +89,13 @@
    <filter>taxonomy_in != None</filter>
   </data>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/get.seqs.xml]: remove?

File: tools/mothur/tools/mothur/heatmap.sim.xml [hunk 8]
```diff
@@ -89,19 +94,13 @@
     <filter>as_datasets != 'true' and input['source'] != 'shared'</filter>
   </data>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/heatmap.sim.xml]: remove

File: tools/mothur/tools/mothur/list.seqs.xml [hunk 9]
```diff
@@ -59,19 +64,13 @@
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
   <data format="accnos" name="out_file" label="${tool.name} on ${on_string}: accnos" />
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/list.seqs.xml]: remove

File: tools/mothur/tools/mothur/make.shared_from_biom.xml [hunk 10]
```diff
@@ -30,19 +35,13 @@
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
   <data format="shared" name="shared" label="${tool.name} on ${on_string}: shared" />
  </outputs>
- <requirements>
   <requirement type="package" version="1.27">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/make.shared_from_biom.xml]: remove

File: tools/mothur/tools/mothur/parse.list.xml [hunk 11]
```diff
@@ -30,19 +35,13 @@
  <outputs>
   <data format="html" name="logfile" label="${tool.name} on ${on_string}: logfile" />
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/parse.list.xml]: remove

File: tools/mothur/tools/mothur/parsimony.xml [hunk 12]
```diff
@@ -50,19 +55,13 @@
   </data>
   -->
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/parsimony.xml]: remove

File: tools/mothur/tools/mothur/phylo.diversity.xml [hunk 13]
```diff
@@ -77,19 +82,14 @@
   </data>
   -->
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/phylo.diversity.xml]: remove

File: tools/mothur/tools/mothur/remove.lineage.xml [hunk 14]
```diff
@@ -96,19 +101,13 @@
    <filter>alignreport_in != None</filter>
   </data>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/remove.lineage.xml]: remove

File: tools/mothur/tools/mothur/remove.rare.xml [hunk 15]
```diff
@@ -94,19 +99,13 @@
    <filter>input['source'] == 'list'</filter>
   </data>
  </outputs>
- <requirements>
 <requirement type = "package" version ="1.33">mothur</requirement>
```
Comment (blankenberg) [inline tools/mothur/tools/mothur/remove.rare.xml]: remove. I'm going to stop marking from here, but the remaining requirements that have also been replaced by macros should be removed as well.

**Comment by blankenberg** [comment]
@bgruening I'm ok if you want to merge and then address the other issues.

**Comment by bgruening** [comment]
@blankenberg ok. I think we should merge the PR soon so prevent merge conflicts. This will also give others the chance to help @oxyko.

**Pull request merged**